src/iarc7_motion/task_command_handler.py

```diff
@@ -108,28 +111,57 @@ def _get_task_command(self):
             try:
                 task_request = self._task.get_desired_command()
             except Exception as e:
-                rospy.logerr('Exception getting tasks preferred velocity')
+                rospy.logerr('Exception getting task command')
                 rospy.logerr(str(e))
                 rospy.logerr(traceback.format_exc())
                 rospy.logerr('Task Command Handler aborted task')
                 self._task = None
                 self._task_state = task_states.TaskAborted()
                 return (task_commands.NopCommand(),)
 
-            self._task_state = task_request[0]
+            try: 
+                _task_state = task_request[0]
+
+                if issubclass(_task_state, task_states.TaskState):
+                    self._task_state = _task_state
+                else: 
+                    rospy.logerr('Task provided unknown state')
+                    rospy.logerr('Task Command Handler aborted task')
+                    self._task = None
+                    self._task_state = task_states.TaskAborted(msg='Error getting task state')
+                    return (task_commands.NopCommand(),)
 
-            if isinstance(self._task_state, task_states.TaskDone):
+            except (TypeError, IndexError) as e: 
```

Comment:
Move this block up to line 124, no need to keep lines 125-132 inside the try block

Reply:
fixed